app/app_test.go

```diff
@@ -261,275 +269,102 @@ func TestDifferentTxForms(t *testing.T) {
-	var addrs [6]common.Address
+	var addrs []common.Address
 
 	for i := 0; i < 6; i++ {
 		keys[i], _ = ethcrypto.GenerateKey()
-		addrs[i] = utils.PrivKeyToAddress(keys[i])
+		addrs = append(addrs, utils.PrivKeyToAddress(keys[i]))
 	}
 
-	InitTestChain(addrs[0], cc)
-	cc.Commit()
-
-	// Create confirm signature
-	confirmSig1 := CreateConfirmSig(types.NewPosition(0, 0, 0, 1), keys[0], &ecdsa.PrivateKey{}, false)
-
-	// Create first tx, 1 input 2 output
-	// Block 7
-	msg := types.SpendMsg{
-		Blknum1:      0,
-		Txindex1:     uint16(0),
-		Oindex1:      uint8(0),
-		DepositNum1:  1,
-		Owner1:       addrs[0],
-		ConfirmSigs1: confirmSig1,
-		Blknum2:      0,
-		Txindex2:     0,
-		Oindex2:      0,
-		DepositNum2:  0,
-		Owner2:       common.Address{},
-		ConfirmSigs2: [2]types.Signature{types.Signature{}, types.Signature{}},
-		Newowner1:    addrs[1],
-		Denom1:       20,
-		Newowner2:    addrs[2],
-		Denom2:       80,
-		Fee:          0,
-	}
-
-	// Sign the hash of the transaction
-	tx := GetTx(msg, keys[0])
-
-	cc.BeginBlock(abci.RequestBeginBlock{Header: abci.Header{Height: 7}})
-
-	// Run a check
-	cres := cc.Check(tx)
-	require.Equal(t, sdk.CodeType(0), sdk.CodeType(cres.Code), cres.Log)
-
-	dres := cc.Deliver(tx)
-
-	require.Equal(t, sdk.CodeType(0), sdk.CodeType(dres.Code), dres.Log)
-
-	// Create context
-	ctx := cc.NewContext(false, abci.Header{})
-
-	// Retrieve UTXO from context
-	utxo1 := cc.utxoMapper.GetUTXO(ctx, addrs[1], types.NewPosition(7, 0, 0, 0))
-	expected1 := types.NewBaseUTXO(addrs[1], [2]common.Address{addrs[0], common.Address{}}, 20, types.NewPosition(7, 0, 0, 0))
-	utxo2 := cc.utxoMapper.GetUTXO(ctx, addrs[2], types.NewPosition(7, 0, 1, 0))
-	expected2 := types.NewBaseUTXO(addrs[2], [2]common.Address{addrs[0], common.Address{}}, 80, types.NewPosition(7, 0, 1, 0))
-
-	require.Equal(t, expected1, utxo1, "First UTXO did not get added to store correctly")
-	require.Equal(t, expected2, utxo2, "Second UTXO did not get added to store correctly")
-
-	utxo1 = cc.utxoMapper.GetUTXO(ctx, addrs[0], types.NewPosition(0, 0, 0, 1))
-	require.Nil(t, utxo1, "Deposit was not removed from utxo store")
-
-	cc.EndBlock(abci.RequestEndBlock{})
+	InitTestChain(addrs, cc)
 	cc.Commit()
 
-	// 2 different inputs 1 output
-	// Block 8
-	confirmSig1 = CreateConfirmSig(types.NewPosition(7, 0, 0, 0), keys[0], &ecdsa.PrivateKey{}, false)
-	confirmSig2 := CreateConfirmSig(types.NewPosition(7, 0, 1, 0), keys[0], &ecdsa.PrivateKey{}, false)
-
-	msg = types.SpendMsg{
-		Blknum1:      7,
-		Txindex1:     uint16(0),
-		Oindex1:      uint8(0),
-		DepositNum1:  0,
-		Owner1:       addrs[1],
-		ConfirmSigs1: confirmSig1,
-		Blknum2:      7,
-		Txindex2:     uint16(0),
-		Oindex2:      uint8(1),
-		DepositNum2:  0,
-		Owner2:       addrs[2],
-		ConfirmSigs2: confirmSig2,
-		Newowner1:    addrs[3],
-		Denom1:       100,
-		Newowner2:    common.Address{},
-		Denom2:       0,
-		Fee:          0,
+	cases := []struct {
+		owner_index1   uint64
```

Comment:
I think it would be cleared if there were Address instead of uint64

Reply:
This is useful for signing with the correct keys.